lib/pure/batteries.nim

```diff
@@ -0,0 +1,57 @@
+#
+#
+#            Nim's Runtime Library
+#        (c) Copyright 2012 Andreas Rumpf
+#
+#    See the file "copying.txt", included in this
+#    distribution, for details about the copyright.
+#
+
+when defined(nimdoc) and isMainModule:
+  from std/compileSettings import nil
+  when compileSettings.querySetting(compileSettings.SingleValueSetting.projectFull) == currentSourcePath:
+    ## This is a module that simply imports common modules for your convenience:
+    ##
+    ## .. code-block:: nim
+    ##   import batteries
```

Comment:
It is fine since it is pre-existing, just fyi markdown syntax works and should be preferred in the future.

````
```nim
import batteries
```
````

Reply:
> New modules should be put at `std/` directory so that it enforces std prefix when importing them.

I just placed it where the current prelude.nim file is. I don't really know the difference between /pure and /std. I can move batteries.nim to /std if you think that is better and I can also move prelude.nim if you think that also makes sense. What would you prefer?

Reply:
> It is fine since it is pre-existing, just fyi markdown syntax works and should be preferred in the future.
> 
> ```
> ```nim
> import batteries
> ```
> ```

I just copied the code from prelude.nim, but I can change it if you think that is better. Let me know what you prefer me to do.

Reply:
I've made the requested changes.